lib/html-proofer/middleware.rb

```diff
@@ -0,0 +1,66 @@
+
+
+module HTMLProofer
+  class Middleware
+
+    def self.options
+      @options ||= {
+        type:                :file,
+        allow_missing_href:  true, # Permitted in html5
+        allow_hash_href:     true,
+        check_external_hash: true,
+        check_html:          true,
+        url_ignore:          [/.*/], # Don't try to check local files exist
+      }
+    end
+
+    def initialize(app)
+      @app = app
+    end
+
+    HTML_SIGNATURE = [
+      '<!DOCTYPE HTML',
+      '<HTML',
+      '<HEAD',
+      '<SCRIPT',
+      '<IFRAME',
+      '<H1',
+      '<DIV',
+      '<FONT',
+      '<TABLE',
+      '<A',
+      '<STYLE',
+      '<TITLE',
+      '<B',
+      '<BODY',
+      '<BR',
+      '<P',
+      '<!--'
+    ]
```

Comment:
What is this subset based on? Is there a more definite way to get a list (eg., does Nokogiri provide a constant of tag names? What happens if a page has an `<h2` instead of `<h1`?)

Reply:
This is drawn from the golang mime-type sniffing package. 

Almost every page is going to start with `<!DOCTYPE HTML` or a comment.

Reply:
Note that it's checking for 'starts with', not 'contains'. 

If your entire document starts with an `<h2` it's not going to fire, but I think at that point you already know your html is invalid.